### ISSUE TITLE
caffe2 framework installation

### ISSUE THREAD

**Issue by aambiguous** [pytorch/pytorch#47553]
i try to install caffe2 as per the instruction given in document but get these log
```
[2228/3734] cmd.exe /C "cd /D C:\python\pytorch && C:\python\python.exe -m tools.codegen.gen --source-path C:/python/pytorch/cmake/../aten/src/ATen --install_dir C:/python/pytorch/build/aten/src/ATen"
[2229/3734] cmd.exe /C "cd /D C:\python\pytorch\build\caffe2 && C:\python\python.exe C:/python/pytorch/caffe2/contrib/aten/gen_op.py --aten_root=C:/python/pytorch/caffe2/../aten --template_dir=C:/python/pytorch/caffe2/contrib/aten --yaml_dir=C:/python/pytorch/build/caffe2/../aten/src/ATen --install_dir=C:/python/pytorch/build/caffe2/contrib/aten"
FAILED: caffe2/contrib/aten/aten_op.h
cmd.exe /C "cd /D C:\python\pytorch\build\caffe2 && C:\python\python.exe C:/python/pytorch/caffe2/contrib/aten/gen_op.py --aten_root=C:/python/pytorch/caffe2/../aten --template_dir=C:/python/pytorch/caffe2/contrib/aten --yaml_dir=C:/python/pytorch/build/caffe2/../aten/src/ATen --install_dir=C:/python/pytorch/build/caffe2/contrib/aten"
Traceback (most recent call last):
  File "C:/python/pytorch/caffe2/contrib/aten/gen_op.py", line 40, in <module>
    from tools.codegen.code_template import CodeTemplate as CT
ModuleNotFoundError: No module named 'tools.codegen'
[2230/3734] cmd.exe /C "cd /D C:\python\pytorch && C:\python\python.exe tools/setup_helpers/generate_code.py --declarations-path C:/python/pytorch/build/aten/src/ATen/Declarations.yaml --native-functions-path aten/src/ATen/native/native_functions.yaml --nn-path aten/src   --force_schema_registration"
Skipped writing torch/csrc\autograd\generated\python_functions.h
Skipped writing torch/csrc\autograd\generated\python_functions.cpp
Skipped writing torch/csrc\autograd\generated\python_variable_methods.cpp
Skipped writing torch/csrc\autograd\generated\python_torch_functions.cpp
Skipped writing torch/csrc\autograd\generated\python_nn_functions.cpp
Skipped writing torch/csrc\autograd\generated\python_fft_functions.cpp
Skipped writing torch/csrc\autograd\generated\python_linalg_functions.cpp
Skipped writing torch/csrc\autograd\generated\VariableType.h
Skipped writing torch/csrc\autograd\generated\VariableType_0.cpp
Skipped writing torch/csrc\autograd\generated\TraceType_0.cpp
Skipped writing torch/csrc\autograd\generated\VariableType_1.cpp
Skipped writing torch/csrc\autograd\generated\TraceType_1.cpp
Skipped writing torch/csrc\autograd\generated\VariableType_2.cpp
Skipped writing torch/csrc\autograd\generated\TraceType_2.cpp
Skipped writing torch/csrc\autograd\generated\VariableType_3.cpp
Skipped writing torch/csrc\autograd\generated\TraceType_3.cpp
Skipped writing torch/csrc\autograd\generated\VariableType_4.cpp
Skipped writing torch/csrc\autograd\generated\TraceType_4.cpp
Skipped writing torch/csrc\autograd\generated\VariableTypeEverything.cpp
Skipped writing torch/csrc\autograd\generated\TraceTypeEverything.cpp
Skipped writing torch/csrc\autograd\generated\Functions.h
Skipped writing torch/csrc\autograd\generated\Functions.cpp
Skipped writing torch/csrc\autograd\generated\variable_factories.h
Skipped writing torch/csrc\jit\generated\generated_unboxing_wrappers_0.cpp
Skipped writing torch/csrc\jit\generated\generated_unboxing_wrappers_1.cpp
Skipped writing torch/csrc\jit\generated\generated_unboxing_wrappers_2.cpp
Skipped writing torch/csrc\jit\generated\generated_unboxing_wrappers_everything.cpp
Skipped writing torch/testing/_internal/generated\annotated_fn_args.py
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "C:\python\pytorch\tools\build_libtorch.py", line 22, in <module>
    rerun_cmake=True, cmake_only=False, cmake=CMake())
  File "C:\python\pytorch\tools\build_pytorch_libs.py", line 62, in build_caffe2
    cmake.build(my_env)
  File "C:\python\pytorch\tools\setup_helpers\cmake.py", line 346, in build
    self.run(build_args, my_env)
  File "C:\python\pytorch\tools\setup_helpers\cmake.py", line 141, in run
    check_call(command, cwd=self.build_dir, env=env)
  File "C:\python\lib\subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install', '--config', 'Release', '--', '-j', '4']' returned non-zero exit status 1.
"Caffe2 building failed"
```
cc @malfet @seemethere @walterddr @ezyang @bhosmer @smessmer @ljk53 @bdhirsh @ailzhang @gchanan @zou3519

**Comment by seemethere** [comment]
Removing the high priority label since this is labeled as `caffe2` and `caffe2` has been deprecated for a while.

I'm not entirely sure I understand the process of how you got here?

Can you provide more details about what happened?

**Comment by xunleiwu** [comment]
I have the same issue with the master branch. Here is my environment
conda 4.9.2
Python 3.8.3
cuda 10.2
cudnn 8.0.2
nccl 2.6.4
gcc (GCC) 7.4.0
RHEL 7.8 with 3.10.0-1127.13.1.el7.x86_64

**Comment by Spacefish** [comment]
Same issue here.
As far as i can see, it changes the current directory into the caffe2 module and then does not find the tools/codegen module as this is located in pytorch and not inside caffee2

**Comment by ezyang** [comment]
Oops, we broke this. Probably the easiest hotfix is to add the PYTorch root dir to os.path. Would accept a PR that does this, add me as reviewer.

**Comment by seemethere** [comment]
Looks like this issue should be resolved by #50353, going to close.

Feel free to re-open if you feel as though this is still an issue.
